File: Programmers/짝지어_제거하기.cpp
#include <iostream>
#include<string>
#include<stack>
using namespace std;

int solution(string s)
{
	int answer = 0;
	stack<char> st;
	for (int i = 0; i < s.length(); i++) {
		if (st.empty() || st.top() != s[i]) st.push(s[i]);
		else st.pop();
	}
	answer = (st.empty()) ? 1 : 0;
	return answer;
}

File: Programmers/java_practice/짝지어_제거하기.java
import java.util.*;

class Solution {
	public int solution(String s) {
		int answer = 0;
		Stack<Character> stack = new Stack<>();
		for (char c : s.toCharArray()) {
			if (stack.empty() || stack.peek() != c)
				stack.push(c);
			else
				stack.pop();
		}
		answer = (stack.empty()) ? 1 : 0;
		return answer;
	}
1}